lib/elixir/src/elixir_with.erl

```diff
@@ -57,3 +80,33 @@ build_cases([], DoExpr) ->
   DoExpr;
 build_cases([{_, Meta, _} = Expr | Rest], DoExpr) ->
   {'__block__', Meta, [Expr, build_cases(Rest, DoExpr)]}.
+
+build_else(Meta, WithCases, RefVar, ElseClauses) ->
+  Result = {'result', Meta, nil},
+  CatchAllExist = catch_all_exists(ElseClauses),
+  Clauses = [
+    {'->', Meta, [[{{'^', Meta, [RefVar]}, Result}], Result]}
+    | ElseClauses
+  ] ++ [build_raise(Meta) || not CatchAllExist],
+  {'case', Meta, [WithCases, [{do, Clauses}]]}.
+
+catch_all_exists([{'->', _, [[Match], _]} | ElseClauses]) ->
```

Comment:
I think it would be beneficial to rename `catch_all_exists` (maybe `has_wildcard_clause`?) as it could be confused with `try`'s catch.
For a second I had that confusion when was reading this https://github.com/elixir-lang/elixir/pull/4114/files#r48451731 comment.